### ISSUE TITLE
Rolledback from 1.7.5 to 1.7.4.4 from hosting backup and get ServiceNotFoundException

### ISSUE THREAD

**Issue by pvechi** [PrestaShop/PrestaShop#12843]
Updated store from Prestashop 1.7.4.4 to 1.7.5.1 successfully using latest version of 1-Click Upgrade.  The upgrade went well.

We then updated the template but found a bug, so we had to rollback to 1.7.4.4.  We did this using our hosting backup files which we rolled back on other occasions and updates with success.

After rolling back from the server backups, the FO works fine.

The back office we cannot even reach login page as it gives us the following error:

`
(1/1) ServiceNotFoundExceptionThe service "prestashop.module.link_block.grid.data_factory" has a dependency on a non-existent service "prestashop.core.hook.dispatcher".
--
in CheckExceptionOnInvalidReferenceBehaviorPass.php line 31
at CheckExceptionOnInvalidReferenceBehaviorPass->processValue(object(Reference), false)in AbstractRecursivePass.php line 60
at AbstractRecursivePass->processValue(array('$gridQueryBuilder' => object(Reference), '$hookDispatcher' => object(Reference), '$queryParser' =>object(Reference), '$gridId' => 'link_block'), false)in CheckExceptionOnInvalidReferenceBehaviorPass.php line 28
at CheckExceptionOnInvalidReferenceBehaviorPass->processValue(array('$gridQueryBuilder' => object(Reference), '$hookDispatcher' => object(Reference), '$queryParser' => object(Reference), '$gridId' => 'link_block'))in AbstractRecursivePass.php line 67

`

Isn't the prestashop.module.link_block.grid.data_factory part of 1.7.5 and not 1.7.4?

See this link: https://devdocs.prestashop.com/1.7/development/components/grid/

Any ideas of what we can to do resolve?  Maybe delete cache objects or directories?

**Additionnal information**
PrestaShop version: 1.7.4.4 rollback from 1.7.5.1
PHP version: 7.1

**Comment by matks** [comment]
This is actually a Symfony service configured into the ps_linklist module: https://github.com/PrestaShop/ps_linklist/blob/master/config/services.yml#L26

Did your update upgrade your modules too ? Maybe you have the upgraded ps_linklist module although you went back to 1.7.4.4.

**Comment by pvechi** [comment]
Thanks for your quick reply. The link block module we updated manually from Installed Modules > Updates.  We updated together with the 1-Click auto update suggested under the updates notices under Modules prior to going to 1.7.5.1.  The updates were carried out after the backups.

**Comment by pvechi** [comment]
I see that the config folder for ps_linklist has a date similar to when the module was upgraded.  Could that be the cause?

**Comment by khouloudbelguith** [comment]
Hi @pvechi,

What is the exact version of the module ps_linklist did you use now?
Thanks!

**Comment by pvechi** [comment]
Hi @khouloudbelguith and @matks,
I think I fixed the problem.  I checked all the backups and none of them have the ps_linklist/config folder. But even though I restored the backup the folder remained in the module's directory.  After removing that folder, I was able to access the BO.  So problem is solved and thanks for pointing us in the right direction here.  Seems to be the hosting's restore backup process.  In any case, the version of LinkList that we have installed on PS 1.7.4.4 is v2.1.5.  It is requesting an update to v2.1.6 under Module Notifications.
